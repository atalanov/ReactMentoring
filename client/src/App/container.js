import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router';
import Component from './component';
import actions from '../actions/filter-action';

const mapStateToProps = state => ({
    filter: state.reducer.filterReducer.filter,
});

function mapDispatchToProps(dispatch) {
    return {
        filterActions: bindActionCreators(actions, dispatch),
    };
}

class App extends React.Component {
    render() {
        return (<Component>
          {this.props.children}
        </Component>);
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
