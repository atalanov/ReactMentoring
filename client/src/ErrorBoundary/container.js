import React from 'react';
import { Logger as logger } from './utils';

class ErrorBoundary extends React.Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
    componentDidCatch(error, info) {
      this.setState({ hasError: true });
      logger.log(error, info);
    }
    render() {
      if (this.state.hasError) {
        // You can render any custom fallback UI
        return <h1>Something went wrong.</h1>;
      }
      return this.props.children;
    }
  }
