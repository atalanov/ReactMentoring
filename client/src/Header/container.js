import React from 'react';
import { connect } from 'react-redux';
import Component from './component';

const mapStateToProps = state => ({
    filter: state.reducer.filterReducer.filter,
});
export class Header extends React.Component {
    handleClick() {
        this.props.actions.get(this.props.filter);
    }
    render() {
        return (
          <Component filter={this.props.filter} />
        );
    }
}

export default connect(mapStateToProps)(Header);
