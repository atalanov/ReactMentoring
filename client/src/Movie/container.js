// @flow
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import actions from '../actions/filter-action';
import { events } from '../constants/constants';
import Component from './component';

const mapStateToProps = state => ({
    movie: state.reducer.filterReducer.movie,
    pending: state.reducer.filterReducer.pending,
});

function mapDispatchToProps(dispatch) {
    return {
        filterActions: bindActionCreators(actions, dispatch),
    };
}

export class Movie extends React.Component {
    componentDidMount() {
        this.props.filterActions.getByID(this.props.match.params.id);
    }
    render() {
        console.log(this.props.movie);
        if (this.props.movie) {
            return (<Component item={this.props.movie} />);
        }
        if (this.props.pending) {
            return (<div>Pending... Please stand by... </div>);
        }
        return (<div />);
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);
