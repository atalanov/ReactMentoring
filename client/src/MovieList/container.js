import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import MoviePreview from '../MoviePreview';
import actions from '../actions/filter-action';
import Component from './component';
import { events } from '../constants/constants';

const mapStateToProps = state => ({
    previews: state.reducer.filterReducer.previews.data,
    filter: state.reducer.filterReducer.filter,
    pendingEvent: state.reducer.filterReducer.pendingEvent,
    pending: state.reducer.filterReducer.pending,
});

export class MovieList extends React.Component {
    render() {
        if (this.props.previews && this.props.previews.length) {
            const listPreviews = this.props.previews.map(item => (<MoviePreview key={item.id} item={item} />));
            return (<Component list={listPreviews} />);
        }
        if (this.props.pending) {
            return (<div>Pending... Please stand by... </div>);
        }
        return (<div />);
    }
}

export default connect(mapStateToProps)(MovieList);
