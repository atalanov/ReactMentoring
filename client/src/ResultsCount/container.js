import React from 'react';
import PropTypes from 'prop-types';
import ResultsCountComponent from './component';
import utils from './utils';

export default class ResultsCount extends React.Component {
    constructor(props) {
        super(props);
        this.countText = utils.formatCount(props.count);
    }
    render() {
        return (
          <ResultsCountComponent countText={this.countText} />
        );
    }
}
