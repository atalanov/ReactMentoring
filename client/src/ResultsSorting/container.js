import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ResultSortingComponent from './component';
import action from '../actions/filter-action';

const mapStateToProps = state => ({
    filter: state.reducer.filterReducer.filter,
});

function mapDispatchToProps(dispatch) { 
    return {
        actions: bindActionCreators(action, dispatch)
    }
}

export class ResultSorting extends React.Component {
    constructor(props) {
        super(props);
        this.values = ['release_date', 'rating'];
        this.selected = 0;
    }
    handleClick = (index) => {
        this.selected = index;
        this.props.actions.changeFilterParam({
            sortBy: this.values[index],
        });
    }
    render() {
        const switcher = this.values.map((item, index) => <ResultSortingComponent key={item} selected={this.selected === index} text={item} handleClick={() =>this.handleClick(index)} />);
        return (
          <div>
            Sort by {switcher}
          </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ResultSorting);
