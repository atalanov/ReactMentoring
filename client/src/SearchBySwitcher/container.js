import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchByComponent from './component';
import action from '../actions/filter-action';

const mapStateToProps = state => ({
    filter: state.reducer.filterReducer.filter,
});

function mapDispatchToProps(dispatch) { 
    return {
        actions: bindActionCreators(action, dispatch)
    }
}
class SearchBySwitcher extends React.Component {
    constructor(props) {
        super(props);
        this.values = ['title', 'genres'];
        this.selected = 0;
    }
    handleClick = (index) => {
        this.selected = index;
        this.props.actions.changeFilterParam({
            searchBy: this.values[index],
        });
    }
    render() {
        const switcher = this.values.map((item, index) => <SearchByComponent key={item} selected={this.selected === index} text={item} handleClick={() =>this.handleClick(index)} />);
        return (
          <div>
            Search by {switcher}
          </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBySwitcher);
