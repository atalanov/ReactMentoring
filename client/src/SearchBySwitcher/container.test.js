import React from 'react';
import { shallow, render } from 'enzyme';
import { SearchBySwitcher } from './container';

describe('SearchBySwitcher Tests', () => {
    test('Component SearchBySwitcher renders correctly', () => {
        const component = shallow(<SearchBySwitcher />);
        expect(component).toMatchSnapshot();
    });
    test('Component SearchBySwitcher default selected value is title', () => {
        const component = render(<SearchBySwitcher />);
        expect(component.find('button:first-child').hasClass('selected')).toBe(true);
    });
});
