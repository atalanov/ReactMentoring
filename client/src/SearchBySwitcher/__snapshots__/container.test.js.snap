// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBySwitcher Tests Component SearchBySwitcher renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBySwitcher />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Search by ",
        Array [
          <Unknown
            selected={true}
            text="title"
          />,
          <Unknown
            selected={false}
            text="genre"
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      "Search by ",
      Object {
        "instance": null,
        "key": "title",
        "nodeType": "function",
        "props": Object {
          "selected": true,
          "text": "title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "genre",
        "nodeType": "function",
        "props": Object {
          "selected": false,
          "text": "genre",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Search by ",
          Array [
            <Unknown
              selected={true}
              text="title"
            />,
            <Unknown
              selected={false}
              text="genre"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        "Search by ",
        Object {
          "instance": null,
          "key": "title",
          "nodeType": "function",
          "props": Object {
            "selected": true,
            "text": "title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "genre",
          "nodeType": "function",
          "props": Object {
            "selected": false,
            "text": "genre",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
